components/notion/actions/append-block/append-block.mjs

```diff
@@ -61,13 +60,17 @@ export default {
         children.push(...blockChildren);
       }
     }
-    blocks.push(...children
-      .filter((child) => Object.keys(child[child.type]).length > 0 && child.type !== "child_page")
-      .map((child) => ({
-        object: "block",
-        type: child.type,
-        [child.type]: child[child.type],
-      })));
+
+    const blocks = [];
+    for (const block of children) {
+      if (!(Object.keys(block[block.type])?.length > 0)) {
+        continue;
+      }
+
+      const formattedChildBlocks = await this.formatChildBlocks(block);
+      const child = this.createChild(block, formattedChildBlocks);
+      blocks.push(child);
+    }
```

Comment:
I think this code can be moved inside `common/base-page-builder.mjs` so `duplicate-page.mjs` component can use it from there too don't you think?